xarray/tests/test_cftimeindex_resample.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import absolute_import
+
+import pytest
+
+import datetime
+import numpy as np
+import pandas as pd
+import xarray as xr
+from xarray.core.resample_cftime import CFTimeGrouper
+
+pytest.importorskip('cftime')
+pytest.importorskip('pandas', minversion='0.23.99')
+
+
+@pytest.fixture(
+    params=[
+        dict(start='2004-01-01T12:07:01', periods=91, freq='3D'),
+        dict(start='1892-01-03T12:07:01', periods=15, freq='41987T'),
+        dict(start='2004-01-01T12:07:01', periods=7, freq='3Q-AUG'),
+        dict(start='1892-01-03T12:07:01', periods=10, freq='3AS-JUN')
+    ],
+    ids=['3D', '41987T', '3Q_AUG', '3AS_JUN']
+)
+def time_range_kwargs(request):
+    return request.param
+
+
+@pytest.fixture()
+def datetime_index(time_range_kwargs):
+    return pd.date_range(**time_range_kwargs)
+
+
+@pytest.fixture()
+def cftime_index(time_range_kwargs):
+    return xr.cftime_range(**time_range_kwargs)
+
+
+def da(index):
+    return xr.DataArray(np.arange(100., 100. + index.size),
+                        coords=[index], dims=['time'])
+
+
+@pytest.mark.parametrize('freq', [
+    '700T', '8001T',
+    '12H', '8001H',
+    '8D', '8001D',
+    '2MS', '3MS',
+    '2QS-AUG', '3QS-SEP',
+    '3AS-MAR', '4A-MAY'])
+@pytest.mark.parametrize('closed', [None, 'right'])
+@pytest.mark.parametrize('label', [None, 'right'])
+@pytest.mark.parametrize('base', [12, 31])
+def test_resampler(freq, closed, label, base,
+                   datetime_index, cftime_index):
+    # Fairly extensive testing for standard/proleptic Gregorian calendar
+    # For any frequencies which are not greater-than-day and anchored
+    # at the end, the default values for closed and label are 'left'.
+    loffset = '12H'
+    try:
+        da_datetime = da(datetime_index).resample(
+            time=freq, closed=closed, label=label, base=base,
+            loffset=loffset).mean()
+    except ValueError:
+        with pytest.raises(ValueError):
+            da(cftime_index).resample(
+                time=freq, closed=closed, label=label, base=base,
+                loffset=loffset).mean()
+    else:
+        da_cftime = da(cftime_index).resample(time=freq, closed=closed,
+                                              label=label, base=base,
+                                              loffset=loffset).mean()
+        da_cftime['time'] = da_cftime.indexes['time'].to_datetimeindex()
+        xr.testing.assert_identical(da_cftime, da_datetime)
+
+
+@pytest.mark.parametrize('freq', [
+    '2M', '3M',
+    '2Q-JUN', '3Q-JUL',
```

Comment:
Ah, I see you've made some changes related to resampling with quarterly frequencies here.  My apologies, I didn't notice that before.  That's great, but it would be nice to hold off on that until #2593 has been officially merged.  It looks like most of what you're looking to add in this PR is independent of resampling (you only need to update the definition of `is_super_daily` and how the default arguments are defined, right?).

Reply:
Most of the changes are indeed not resample related, except for `is_super_daily` in get_time_bins and similar code in `CFTimeGrouper`.

I'll start a new pull request that doesn't touch any of the resample code.

Reply:
Sounds great, thanks -- that will make things easier to review 👍.  Sorry for my confusion here.